### PR TITLE
Use latest Gradle wrapper and Bump dependencies

### PR DESCRIPTION
**Pull request by schnapster**

### CODE REVIEW

**Comment by lbalmaceda** [comment]
I forgot to update this one on the past week.. This is good. But please, go and rollback the changes on `spring` and `servlet` dependencies since you are now using different majors and I can't know for sure what's the impact for this SDK's users.

File: lib/build.gradle [hunk 1]
```diff
@@ -31,20 +31,20 @@ compileJava {
...
-    compile 'org.springframework.security:spring-security-core:4.2.4.RELEASE'
-    compile 'org.springframework.security:spring-security-web:4.2.4.RELEASE'
-    compile 'org.springframework.security:spring-security-config:4.2.4.RELEASE'
-    compile 'org.slf4j:slf4j-api:1.7.25'
-    compileOnly 'javax.servlet:servlet-api:2.5'
+    implementation 'com.auth0:java-jwt:3.4.1'
+    implementation 'com.auth0:jwks-rsa:0.6.1'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.9.7'
+    implementation 'commons-codec:commons-codec:1.11'
+    implementation 'org.springframework.security:spring-security-core:5.1.1.RELEASE'
```
Comment (lbalmaceda) [inline lib/build.gradle]:
```suggestion
    implementation 'org.springframework.security:spring-security-core:4.2.9.RELEASE'
```

File: lib/build.gradle [hunk 2]
```diff
@@ -31,20 +31,20 @@ compileJava {
...
+    implementation 'org.springframework.security:spring-security-web:5.1.1.RELEASE'
```
Comment (lbalmaceda) [inline lib/build.gradle]:
```suggestion
    implementation 'org.springframework.security:spring-security-web:4.2.9.RELEASE'
```

File: lib/build.gradle [hunk 3]
```diff
@@ -31,20 +31,20 @@ compileJava {
...
+    implementation 'org.springframework.security:spring-security-config:5.1.1.RELEASE'
```
Comment (lbalmaceda) [inline lib/build.gradle]:
```suggestion
    implementation 'org.springframework.security:spring-security-config:4.2.9.RELEASE'
```

File: lib/build.gradle [hunk 4]
```diff
@@ -31,20 +31,20 @@ compileJava {
...
+    compileOnly 'javax.servlet:javax.servlet-api:4.0.1'
 
-    testCompile 'junit:junit:4.12'
-    testCompile 'org.hamcrest:java-hamcrest:2.0.0.0'
-    testCompile 'org.mockito:mockito-core:2.13.0'
-    testCompile 'javax.servlet:servlet-api:2.5'
+    testImplementation 'junit:junit:4.12'
+    testImplementation 'org.hamcrest:java-hamcrest:2.0.0.0'
+    testImplementation 'org.mockito:mockito-core:2.23.0'
+    testCompile 'javax.servlet:javax.servlet-api:4.0.1'
```
Comment (lbalmaceda) [inline lib/build.gradle]:
```suggestion
    testCompile 'javax.servlet:servlet-api:2.5'
```

File: lib/build.gradle [hunk 5]
```diff
@@ -31,20 +31,20 @@ compileJava {
...
-    compileOnly 'javax.servlet:servlet-api:2.5'
+    implementation 'com.auth0:java-jwt:3.4.1'
+    implementation 'com.auth0:jwks-rsa:0.6.1'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.9.7'
+    implementation 'commons-codec:commons-codec:1.11'
+    implementation 'org.springframework.security:spring-security-core:5.1.1.RELEASE'
+    implementation 'org.springframework.security:spring-security-web:5.1.1.RELEASE'
+    implementation 'org.springframework.security:spring-security-config:5.1.1.RELEASE'
+    implementation 'org.slf4j:slf4j-api:1.7.25'
+    compileOnly 'javax.servlet:javax.servlet-api:4.0.1'
```
Comment (lbalmaceda) [inline lib/build.gradle]:
```suggestion
    compileOnly 'javax.servlet:servlet-api:2.5'
```

**Review by lbalmaceda — CHANGES_REQUESTED** [review]
Added a comment above this review. Added suggested changes to avoid changing a dependency version to a new major.

**Comment by schnapster** [comment]
Gotcha, updated.

File: build.gradle [hunk 6]
```diff
@@ -18,4 +18,8 @@ allprojects {
     repositories {
         jcenter()
     }
-}
\ No newline at end of file
+}
+
+wrapper {
```
Comment (lbalmaceda) [inline build.gradle]: I've never needed this before. Can you safely remove it? How have you updated the gradle version? I normally run `./gradlew wrapper  --gradle-version 4.10.2 --distribution-type all`

**Review by lbalmaceda — APPROVED** [review]

**Pull request merged**